lumen/ai/agents.py

```diff
@@ -127,7 +129,7 @@ def __panel__(self):
 
     async def _stream(self, messages: list[Message], system_prompt: str) -> Any:
         message = None
-        model_spec = self.prompts["main"].get("llm_spec", "default")
+        model_spec = self.prompts["main"].get("llm_spec", self._llm_spec_key)
```

Comment:
It'll still use default if the key isn't found
https://github.com/holoviz/lumen/pull/1100/files#diff-dc864a2206353ab4b7498c9a9361f5c45af0c5fd84adbc59fe054515346e55d4L77-L83